ghost/core/test/regression/models/model_members.test.js

```diff
@@ -503,6 +505,74 @@ describe('Member Model', function run() {
                 should.equal(members.data.length, 0, 'Can search for members by plan_interval');
             }
         });
+
+        it('Should allow filtering on offers redeemed', async function () {
+            const context = testUtils.context.admin;
+
+            const member = await Member.add({
+                email: 'test@test.member',
+                labels: []
+            }, context);
+
+            const product = await Product.add({
+                name: 'Ghost Product',
+                slug: 'ghost-product',
+                type: 'paid'
+            }, context);
+
+            const offerData = testUtils.DataGenerator.forKnex.createOffer({
+                product_id: product.get('id')
+            });
+            const offerModel = await Offer.add(offerData, {context: {internal: true}});
+
+            await StripeProduct.add({
+                product_id: product.get('id'),
+                stripe_product_id: 'fake_product_id'
+            }, context);
+
+            await StripePrice.add({
+                stripe_price_id: 'fake_plan_id',
+                stripe_product_id: 'fake_product_id',
+                amount: 5000,
+                interval: 'monthly',
+                active: 1,
+                nickname: 'Monthly',
+                currency: 'USD',
+                type: 'recurring'
+            }, context);
+
+            await MemberStripeCustomer.add({
+                member_id: member.get('id'),
+                customer_id: 'fake_customer_id1'
+            }, context);
+
+            const subscription = await StripeCustomerSubscription.add({
+                customer_id: 'fake_customer_id1',
+                subscription_id: 'fake_subscription_id1',
+                plan_id: 'fake_plan_id',
+                stripe_price_id: 'fake_plan_id',
+                plan_amount: 1337,
+                plan_nickname: 'e-LEET',
+                plan_interval: 'year',
+                plan_currency: 'btc',
+                status: 'active',
+                start_date: new Date(),
+                current_period_end: new Date(),
+                cancel_at_period_end: false
+            }, context);
+
+            const addRedemption = await OfferRedemption.add({
+                offer_id: offerModel.get('id'),
+                member_id: member.get('id'),
+                created_at: new Date(),
+                subscription_id: subscription.get('id')
+            }, context);
+
+            const offerId = addRedemption.get('offer_id');
+
+            const members = await Member.findPage({filter: `offer_redemptions:${offerId}`});
+            should.equal(members.data.length, 1, 'Can search for members with offer redemptions');
```

Comment:
I think we should check that the specific member you added in this test is the one that is returned, maybe update the email address to something like `offerredemptionfilteringtest@whatevber.comsdlkcjkl`

Reply:
Done 😊